### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by hagen1778**
Initial set of metrics for controller and metrics provider

### CODE REVIEW

File: controllers/consumer_controller.go [hunk 1]
```diff
@@ -75,27 +75,36 @@ type ConsumerReconciler struct {
 // +kubebuilder:rbac:groups=apps,resources=deployments/status,verbs=get
 
 func (r *ConsumerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	start := time.Now()
+	defer func() {
+		reconcileDuration.WithLabelValues(req.Namespace).Observe(time.Since(start).Seconds())
```
Comment (lwolf) [inline controllers/consumer_controller.go]: I think we should have `consumer.Name` in every metrics instead of the namespace.

**Comment by hagen1778** [comment]
Wasn't able to verify that metrics are actually scraping by Prom( I'll be absent next few days, so I'll appreciate if it's possible to postpone this PR or merge it without me

**Comment by lwolf** [comment]
I try to test it when you're away, otherwise it'll wait for you.

**Comment by lwolf** [comment]
rebased against master and fixed metrics registration.
New metrics weren't available because controller-manager creates it's own prometheus registry.

**Pull request merged**